tests/test-utils/makeRequestProcessor.ts

```diff
@@ -19,8 +22,19 @@ const DEFAULT_OPTS: MakeFunctionOpts = {
    callsNextWithError: false,
    callsNextRoute: false,
    passesErrorToNext: false,
+   returnsResolvedPromise: false,
+   returnsRejectedPromise: false,
+   returnsEmptyRejectedPromise: false,
 };
 
+function delay<T>(value: T, ms: number): Promise<T> {
```

Comment:
Technically this function is not typed correctly. In the case where `T` is a `Promise`, the return value comes out to `Promise<Promise<U>>` rather than what you'd expect. So, when you add a `then` handler, the type of the parameter is wrong:

```
delay(Promise.resolve('test'), 4).then((value) => {
   // value is typed as Promise<string>
   // when using Promise.reject, the type is Promise<never>
});
```

The easiest thing to do would be to remove the generic `<T>` and just use `Promise<unknown>` since we're not really caring about the return type here. Otherwise, you could try to fix the return type to return the correct type when when T is a Promise. Not sure if that's worth the work though. 🤷‍♂️